dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -137,69 +226,27 @@ PrehandleResult KVStore::preHandleSSTsToDTFiles(
                     context.getSettingsRef().safe_point_update_interval_seconds);
             }
             physical_table_id = storage->getTableInfo().id;
-            auto log_prefix = fmt::format("keyspace={} table_id={}", keyspace_id, physical_table_id);
 
-            auto & global_settings = context.getGlobalContext().getSettingsRef();
+            auto opt = DM::SSTFilesToBlockInputStreamOpts{
+                .log_prefix = fmt::format("keyspace={} table_id={}", keyspace_id, physical_table_id),
+                .schema_snap = schema_snap,
+                .gc_safepoint = gc_safepoint,
+                .force_decode = force_decode,
+                .expected_size = expected_block_size};
 
-            // Read from SSTs and refine the boundary of blocks output to DTFiles
             auto sst_stream = std::make_shared<DM::SSTFilesToBlockInputStream>(
-                log_prefix,
                 new_region,
                 index,
                 snaps,
                 proxy_helper,
-                schema_snap,
-                gc_safepoint,
-                force_decode,
                 tmt,
-                expected_block_size);
-            auto bounded_stream = std::make_shared<DM::BoundedSSTFilesToBlockInputStream>(
-                sst_stream,
-                ::DB::TiDBPkColumnID,
-                schema_snap);
-            stream = std::make_shared<DM::SSTFilesToDTFilesOutputStream<DM::BoundedSSTFilesToBlockInputStreamPtr>>(
-                log_prefix,
-                bounded_stream,
-                storage,
-                schema_snap,
-                job_type,
-                /* split_after_rows */ global_settings.dt_segment_limit_rows,
-                /* split_after_size */ global_settings.dt_segment_limit_size,
-                region_id,
-                prehandle_task,
-                context);
-
-            stream->writePrefix();
-            stream->write();
-            stream->writeSuffix();
-            if (stream->isAbort())
-            {
-                LOG_INFO(
-                    log,
-                    "Apply snapshot is aborted, cancelling. region_id={} term={} index={}",
-                    region_id,
-                    term,
-                    index);
-                stream->cancel();
-            }
-            prehandle_result.ingest_ids = stream->outputFiles();
-            prehandle_result.stats = PrehandleResult::Stats{
-                .raft_snapshot_bytes = sst_stream->getProcessKeys().total_bytes(),
-                .dt_disk_bytes = stream->getTotalBytesOnDisk(),
-                .dt_total_bytes = stream->getTotalCommittedBytes()};
+                DM::SSTFilesToBlockInputStreamOpts(opt));
```

Comment:
```suggestion
                std::move(opt));
```
I think we don't need this copying

Reply:
Why, opts is later used by `readFromStream`

Reply:
OK, I miss that line